Devscord.DiscordFramework/Framework/Commands/Services/BotCommandsParsingService.cs

```diff
@@ -14,12 +16,30 @@ public BotCommandsParsingService(BotCommandsPropertyConversionService botCommand
         }
 
         public IBotCommand ParseRequestToCommand(Type commandType, DiscordRequest request, BotCommandTemplate template)
+        {
+            var result = GetFilledInstance(commandType, template, x => request.Arguments.First(a => a.Name.ToLowerInvariant() == x.ToLowerInvariant())?.Value);
+            return result;
+        }
+
+        public IBotCommand ParseCustomTemplate(Type commandType, BotCommandTemplate template, Regex customTemplate, string input)
+        {
+            var match = customTemplate.Match(input);
+            if(!this.CustomTemplateIsValid(match, template))
+            {
+                Log.Warning("Custom template {customTemplate} is not valid for {commandName}", customTemplate, template.CommandName);
+                return null;
+            }
+            var result = GetFilledInstance(commandType, template, x => match.Groups.ContainsKey(x) ? match.Groups[x].Value : null);
+            return result;
+        }
+
+        private IBotCommand GetFilledInstance(Type commandType, BotCommandTemplate template, Func<string, string> getValueByName)
         {
             var instance = Activator.CreateInstance(commandType);
             foreach (var property in commandType.GetProperties())
             {
-                var value = request.Arguments.FirstOrDefault(x => x.Name.ToLowerInvariant() == property.Name.ToLowerInvariant())?.Value;
-                if (value == null)
+                var value = getValueByName.Invoke(property.Name);
```

Comment:
nawiązując do porzedniego komentarza - jeśli getValueByName będzie == null to rzuci wyjątek

Reply:
a w jakim przypadku to mogłoby być == null?
może zwracać null, ale func sam w sobie nigdy nie będzie nullem 

masz dwie możliwości
`x => match.Groups.ContainsKey(x) ? match.Groups[x].Value : null`
`x => request.Arguments.FirstOrDefault(a => a.Name.ToLowerInvariant() == x.ToLowerInvariant())?.Value`

a zwracanie null jest niżej sprawdzane z string.IsNullOrWhiteSpace()

Reply:
Faktycznie, źle spojrzałem